File: backup/entities/Destination.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\DestinationRepository;

#[ORM\Entity(repositoryClass: DestinationRepository::class)]
#[ORM\Table(name: 'destination')]
class Destination
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_destination', type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $name = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $location = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $category = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $status = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $routes = null;

    #[ORM\Column(name: 'contact_information', type: 'string', nullable: true)]
    private ?string $contactInformation = null;

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $rating = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\OneToMany(targetEntity: Reservation::class, mappedBy: 'destination')]
    private Collection $reservations;

    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;
        return $this;
    }

    public function getLocation(): ?string
    {
        return $this->location;
    }

    public function setLocation(string $location): self
    {
        $this->location = $location;
        return $this;
    }

    public function getCategory(): ?string
    {
        return $this->category;
    }

    public function setCategory(string $category): self
    {
        $this->category = $category;
        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;
        return $this;
    }

    public function getRoutes(): ?string
    {
        return $this->routes;
    }

    public function setRoutes(?string $routes): self
    {
        $this->routes = $routes;
        return $this;
    }

    public function getContactInformation(): ?string
    {
        return $this->contactInformation;
    }

    public function setContactInformation(?string $contactInformation): self
    {
        $this->contactInformation = $contactInformation;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getRating(): ?int
    {
        return $this->rating;
    }

    public function setRating(?int $rating): self
    {
        $this->rating = $rating;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    /**
     * @return Collection<int, Reservation>
     */
    public function getReservations(): Collection
    {
        return $this->reservations;
    }

    public function addReservation(Reservation $reservation): self
    {
        if (!$this->reservations->contains($reservation)) {
            $this->reservations->add($reservation);
            $reservation->setDestination($this);
        }
        return $this;
    }

    public function removeReservation(Reservation $reservation): self
    {
        if ($this->reservations->removeElement($reservation)) {
            if ($reservation->getDestination() === $this) {
                $reservation->setDestination(null);
            }
        }
        return $this;
    }
}

File: src/main/java/Models/Destination.java
package Models;

import javafx.scene.image.Image;

import java.io.File;
import java.util.Objects;

public class Destination {

    private int id_destination;
    private String name, location, category, status, routes, contact_information, image, description;
    private String imageUrl;
    private int rating;
    private boolean isFavorite;

    public Destination(String location, String name, String category, String status, String routes, String contact_information, String image, String description, int rating) {
        this.location = location;
        this.name = name;
        this.category = category;
        this.status = status;
        this.routes = routes;
        this.contact_information = contact_information;
        this.image = image;
        this.description = description;
        this.rating = rating;
    }

    public Destination(String location, String name, int id_destination, String category, String status, String routes, String contact_information, String image, String description, int rating) {
        this.location = location;
        this.name = name;
        this.id_destination = id_destination;
        this.category = category;
        this.status = status;
        this.routes = routes;
        this.contact_information = contact_information;
        this.image = image;
        this.description = description;
        this.rating = rating;
    }

    public int getId_destination() {
        return id_destination;
    }

    public void setId_destination(int id_destination) {
        this.id_destination = id_destination;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getRoutes() {
        return routes;
    }

    public void setRoutes(String routes) {
        this.routes = routes;
    }

    public String getContact_information() {
        return contact_information;
    }

    public void setContact_information(String contact_information) {
        this.contact_information = contact_information;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    @Override
    public String toString() {
        return "Destination{" +
                "id_destination=" + id_destination +

                ", name='" + name + '\'' +
                ", location='" + location + '\'' +
                ", category='" + category + '\'' +
                ", status='" + status + '\'' +
                ", routes='" + routes + '\'' +
                ", contact_information='" + contact_information + '\'' +
                ", image='" + image + '\'' +
                ", description='" + description + '\'' +
                ", rating=" + rating +
                '}';
    }
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Destination that = (Destination) o;
        return id_destination == that.id_destination;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id_destination);
    }

    //handle favorite
    // Add these methods
    public boolean isFavorite() {
        return isFavorite;
    }

    public void setFavorite(boolean favorite) {
        isFavorite = favorite;
    }
    // Returns a JavaFX Image object
    public Image getImageUrl() {
        String basePath = "/home/fsociety/Downloads/FINALVERSIONSYMFONY/Movable-WebApp-hanipushit/public/uploads/destinations/";

        if (image == null || image.isEmpty()) {
            return getPlaceholderImage();
        }

        try {
            String fullPath = basePath + image;
            File file = new File(fullPath);
            if (!file.exists()) {
                return getPlaceholderImage();
            }
            return new Image(file.toURI().toString());
        } catch (Exception e) {
            return getPlaceholderImage();
        }
    }

    public Image getPlaceholderImage() {
        try {
            return new Image(getClass().getResourceAsStream("/placeholder.png"));
        } catch (Exception e) {
            return null;
        }
    }
}

